### DISCUSSION TITLE
Different results with EPIC and quanTIseq

### DISCUSSION THREAD

**Discussion by kpatel427** [omnideconv/immunedeconv#72] [Q&A]
Hi,

I used both EPIC and quanTIseq methods to get immune cell fractions from bulk RNA-Seq of two tumor samples. I get very different results. I am not sure how to determine which one represents actual cell fractions. Also, why do I get such a difference between two methods?
Any help is appreciated!

Thanks,
Khushbu
<img width="1320" alt="Screen Shot 2021-08-11 at 5 14 35 PM" src="https://user-images.githubusercontent.com/27897154/129104516-8e3af55f-711d-4dfe-83d5-ed8cf59ba727.png">
<img width="1319" alt="Screen Shot 2021-08-11 at 5 14 08 PM" src="https://user-images.githubusercontent.com/27897154/129104535-0cccd1bd-eac4-48bd-886e-7ed7e3fc7f49.png">

**Comment by grst** [comment]
Hi Khushbu, 

the extremely high fraction of "other" cells in quanTIseq seems a bit weird. Maybe @FFinotello can give you more information about how the absolute quantification works and what could potentially cause an over- or under-estimation of "other" cells. 

In our [benchmark paper](https://academic.oup.com/bioinformatics/article/35/14/i436/5529146) we weren't able to systematically evaluate the performance of the "absolute quantification" due to limitations in our simulation strategy and the limited availability of FACS data, so we can't reliably judge which of the methods is more likely to be right. 

If you ignore "other" cells and just compare the relative proportions of the immune cells (e.g. scatter plot), I would expect that you have a higher agreement between the methods. 

Best, 
Gregor

**Reply by kpatel427** [reply]
Hi Gregor,

Thank you for the response. 
I shall compare relative proportions of immune cells and see if that yields me comparable results.

Best,
Khushbu

**Comment by FFinotello** [comment]
Hello @kpatel427! 

May I ask you some more information on how the data were preprocessed and summarized (e.g. counts or TPM), as well as on the parameter settings used for deconvolution?

Thanks,
Francesca

**Reply by kpatel427** [reply]
Hi @FFinotello,

I used TPM (not log transformed) as input to generate these plots, from 2 tumor samples.
These were the exact commands used -
`res_epic <- deconvolute_epic(tpm.df, tumor = TRUE, scale_mrna = FALSE)`
`res_quant <- deconvolute_quantiseq(tpm.df, arrays = FALSE ,tumor = TRUE, scale_mrna = FALSE)`

Thank you for taking a look at this.

Best,
Khushbu

**Comment by FFinotello** [comment]
Hello @kpatel427 

the `scale_mrna` option should be set to `TRUE` (the default) for the analysis of real data, otherwise, the cell-type-specific mRNA bias is not corrected. I am not sure this will solve the issue here, but I would definitely give it a try as a first step. 

May I ask which cancer type is this?

Cheers,
Francesca

**Reply by kpatel427** [reply]
Hi Francesca,

Unfortunately `scale_mrna = TRUE` did not fix it. These are tumor samples from papillary thyroid carcinoma.

Thanks,
Khushbu

**Reply by FFinotello** [reply]
In case of over-estimation of a certain cell type, I would suggest in general to check whether some signature genes correlate with the abundances of this cell type (in this case, "Other" cells). But in this case, I find it nearly impossible with just two samples.

Still, we could confirm a good agreement between quanTIseq estimates of "other"cell abundances and tumor content/purity in different cancer types (see [Supplementary Figure S5c](https://static-content.springer.com/esm/art%3A10.1186%2Fs13073-019-0638-6/MediaObjects/13073_2019_638_MOESM2_ESM.pdf) in quanTIseq paper). 

And, as you see, some samples have a quite high tumor content. Have you tried some orthogonal methods for estimating tumor purity from RNA-seq like [ESTIMATE](https://www.nature.com/articles/ncomms3612) to check whether it is closer to 50% or 90%?

Cheers,
Francesca

**Reply by kpatel427** [reply]
I did try ESTIMATE and this is what it looks like:
<img width="958" alt="Screen Shot 2021-08-12 at 11 21 56 AM" src="https://user-images.githubusercontent.com/27897154/129223466-8eea27ca-607f-4906-b193-ee321f02901e.png">


If I am interpreting this correctly, I think this corroborates what I see with EPIC which means quanTIseq is over-estimating "Other" cells. 

Thanks,
Khushbu

**Reply by FFinotello** [reply]
Yes, by looking at your plot, I'd say that by applying the formula to compute tumor purity

`purity <- cos(0.6049872018 + 0.0001467884 * ESTIMATE$ESTIMATE_score)`

you should end up with an estimate of 50-60%.

Usally EPIC and quanTIseq correlate quite well, but i's not easy to compare only the immune cell types in this case. I see some big discrepancies in terms of CD4 T cells, for instance. You can compute them for quanTIseq by summing up (non-regulatory) CD4 and Tregs... but I see that EPIC is estimating a very large proportion (~30%?).

One thing that comes to my mind, is the possibility that there is an unknown cell type (not present in quanTIseq signature) that might end up in the "other" cell quantification. Maybe it's a good idea to try to identify, among the signature genes, those that have higher expression levels and see whether they are markers for some specific cell types. 

I am sorry I cannot be more helpful in this case.

Good luck,
Francesca

**Reply by kpatel427** [reply]
Thank you Francesca, I will try to look for different markers having high expression. I think that is what is happening with quanTIseq, it is grouping unknown signatures in "other" category.

I am really grateful for your help and appreciate you taking the time to look into this.

Best,
Khushbu

**Reply by FFinotello** [reply]
I am sorry for not being truly helpful in this case. In the past, similar results were due to expression levels provided as counts instead of TPM. But I guess this is not the case.
